tests/fixture/bootstrapmonitor/e2e/e2e_test.go

```diff
@@ -0,0 +1,610 @@
+
+	grantMonitorPermissions(tc, clientset, namespace)
+
+	createdStatefulSet, err := clientset.AppsV1().StatefulSets(namespace).Create(tc.DefaultContext(), statefulSet, metav1.CreateOptions{})
+	require.NoError(tc, err)
+
+	return createdStatefulSet
+}
+
+// getMonitorContainer retrieves the common container definition for bootstrap-monitor containers.
+func getMonitorContainer(name string, args []string) corev1.Container {
+	return corev1.Container{
+		Name:    name,
+		Image:   latestMonitorImage,
+		Command: []string{"./bootstrap-monitor"},
+		Args:    args,
+		Env: []corev1.EnvVar{
+			{
+				Name: "POD_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.name",
+					},
+				},
+			},
+			{
+				Name: "POD_NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.namespace",
+					},
+				},
+			},
+		},
+	}
+}
+
+// grantMonitorPermissions grants the permissions required by the bootstrap-monitor to the namespace's default service account.
+func grantMonitorPermissions(tc tests.TestContext, clientset *kubernetes.Clientset, namespace string) {
+	require := require.New(tc)
+
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "bootstrap-monitor-role-",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"pods"},
+				Verbs:     []string{"get", "create", "watch", "delete"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"pods/log"},
+				Verbs:     []string{"get"},
+			},
+			{
+				APIGroups: []string{"apps"},
+				Resources: []string{"statefulsets"},
+				Verbs:     []string{"patch"},
+			},
+		},
+	}
+	createdRole, err := clientset.RbacV1().Roles(namespace).Create(tc.DefaultContext(), role, metav1.CreateOptions{})
+	require.NoError(err)
+
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "bootstrap-monitor-role-binding-",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "default",
+				Namespace: namespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     createdRole.Name,
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+	_, err = clientset.RbacV1().RoleBindings(namespace).Create(tc.DefaultContext(), roleBinding, metav1.CreateOptions{})
+	require.NoError(err)
+}
+
+// waitForLogOutput streams the logs from the specified pod container until the desired output is found or the context times out.
+func waitForLogOutput(tc tests.TestContext, clientset *kubernetes.Clientset, namespace string, podName string, containerName string, desiredOutput string) {
+	// TODO(marun) Figure out why log output is randomly truncated (not flushed?)
+
+	outputLogLine(tc, fmt.Sprintf(
+		"Logs from: %q container of pod %s.%s (may not be complete)", containerName, namespace, podName))
+	outputLogLine(tc,
```

Comment:
Maybe use `strings.Repeat`? Maybe there's a clever way to do this by fetching the terminal width from `x/term` as well.